File: CodingSamples/Java/Platform-Enterprise/FileStreamTest/met-tour/src/main/java/tourism/SiteStore.java
package tourism;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

public class SiteStore {
    
    public static boolean save(Site site) {
        String doc = site.title + ".store";
        try(var serializer = new ObjectOutputStream(new FileOutputStream(doc))){
            serializer.writeObject(site);
            return true;
        }catch(Exception e){
            return false;
        }
    }

    public static Site load(String name) {
        String doc = name + ".store";
        try(var deserializer = new ObjectInputStream(new FileInputStream(doc))){
            return (Site)deserializer.readObject();
        }catch(Exception e){
            return new Site(name);
        }
    }
}

File: CodingSamples/DotNet/Platform-Enterprise/FileStreamTest/MetTourLib/Tourism/SiteStore.cs
using System.IO.Compression;
using System.Text.Json;

namespace Tourism;

public static class SiteStore
{
    public static bool Save(this Site site)
    {
        string doc = site.Title + ".store";
        try
        {
            using var output = new DeflateStream(new FileStream(doc, FileMode.Create), CompressionMode.Compress);
            JsonSerializer.Serialize(output, site);
            return true;
        }
        catch
        {
            return false;
        }
    }

    public static Site Load(string name)
    {
        string doc = name + ".store";
        try
        {
            using var input = new DeflateStream(new FileStream(doc, FileMode.Open), CompressionMode.Decompress);
            return JsonSerializer.Deserialize<Site>(input);
        }
        catch
        {
            return new Site(name);
        }
    }
}
